### PR TITLE
JDK-8273815: move have_special_privileges to os_posix for POSIX platforms

### PR DESCRIPTION
**Pull request by MBaesken**
We have identical have_special_privileges() for Linux/AIX/BSD so the function is a good candidate to be moved to os_posix.

Thanks, Matthias

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273815](https://bugs.openjdk.java.net/browse/JDK-8273815): move have_special_privileges to os_posix for POSIX platforms

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 3e48d8cf6bab6b9213aa9b506ebcab7ae38d9edc
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5540/head:pull/5540` \
`$ git checkout pull/5540`

Update a local copy of the PR: \
`$ git checkout pull/5540` \
`$ git pull https://git.openjdk.java.net/jdk pull/5540/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5540`

View PR using the GUI difftool: \
`$ git pr show -t 5540`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5540.diff">https://git.openjdk.java.net/jdk/pull/5540.diff</a>

</details>

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]

**Comment by MBaesken** [comment]
Hi Martin, thanks for the approval !
Best regards, Matthias

File: src/hotspot/os/posix/os_posix.cpp [hunk 1]
```diff
@@ -183,6 +183,17 @@ size_t os::lasterror(char *buf, size_t len) {
...
+// Return true if user is running as root.
+bool os::have_special_privileges() {
+  static bool init = false;
+  static bool privileges = false;
+  if (!init) {
+    privileges = (getuid() != geteuid()) || (getgid() != getegid());
+    init = true;
+  }
+  return privileges;
+}
```
Comment (tstuefe) [inline src/hotspot/os/posix/os_posix.cpp:195]: Could be shortened to:
```
bool os::have_special_privileges() {
  static bool privileges = (getuid() != geteuid()) || (getgid() != getegid());
  return privileges;
}
```

**Review by tstuefe — APPROVED** [review]
Looks good. Proposal for shorter code inline (if you take it, I need no re-review).

Cheers, Thomas

File: src/hotspot/os/posix/os_posix.cpp (hunk 1 above)
Comment (MBaesken) [reply src/hotspot/os/posix/os_posix.cpp:190]: Hi Thomas, thanks for the hint - I adjusted the coding.
Best regards, Matthias

Comment (TheRealMDoerr) [reply src/hotspot/os/posix/os_posix.cpp:190]: A static bool which gets evaluated on every call makes no sense!

**Review by TheRealMDoerr — APPROVED** [review]
Forget my comment. Still good.

File: src/hotspot/os/posix/os_posix.cpp (hunk 1 above)
Comment (tstuefe) [reply src/hotspot/os/posix/os_posix.cpp:190]: Now I am confused, what class? This is a global function with a function-local static variable.

**Review by dholmes-ora — APPROVED** [review]
As an equivalent refactoring this is fine.

We could also do the initialization is os::init() but it makes little, if any difference.

Thanks,
David

**Pull request closed without merging**